quickwit-metastore/src/metastore/file_backed_metastore/file_backed_index.rs

```diff
@@ -102,7 +102,17 @@ impl From<FileBackedIndex> for FileBackedIndexV0 {
 
 impl From<FileBackedIndexV0> for FileBackedIndex {
     fn from(index: FileBackedIndexV0) -> Self {
-        Self::new(index.metadata, index.splits)
+        // Override split index_id to support old SplitMetadata format.
+        let index_id = index.metadata.index_id.clone();
```

Comment:
Can we do this in place to avoid some allocations? Can we do this only when `index_id` is empty?